pilot/pkg/model/context.go

```diff
@@ -81,6 +81,11 @@ type Proxy struct {
 	// "default.svc.cluster.local")
 	Domain string
 
+	// ConfigNamespace defines the namespace where this proxy resides
+	// for the purposes of network scoping.
+	// NOTE: DO NOT USE THIS FIELD TO CONSTRUCT DNS NAMES
+	ConfigNamespace string
```

Comment:
I dont want to call this "namespace". It would be no time before some clueless soul comes in and uses this to construct the DNS names. I have caught so many people trying to do this. The WorkloadNamespace and ConfigNamespace are okay, but not NamespaceScope. Once again, the choice of ConfigNamespace over workloadNamespace was because of the clueless folks who come in every now and then and use this field in DNS names. 

Also, by ConfigNamespace, it does not imply the namespace where configuration for a pod can be obtained. Its simply a namespace to which the pod has been assigned to, so that when writing configuration, we can refer to the entire group by using the grouping name (which is called configNamespace).